build.gradle

```diff
@@ -216,9 +216,39 @@ task distNativeSources(type: Zip) {
     }
 }
 
+task distEndorsedCommonSources(type: Zip) {
+    dependsOn ":endorsedLibraries:endorsedLibsCommonSources"
+
+    destinationDirectory = file("$distDir/sources")
+    archiveFileName = "kotlin-endorsed-common-sources.zip"
+
+    includeEmptyDirs = false
+    include('**/*.kt')
+
+    from(project(':endorsedLibraries').file('build/sources-common'))
+}
+
+task distEndorsedNativeSources(type: Zip) {
+    dependsOn ":endorsedLibraries:endorsedLibsNativeSources"
+
+    destinationDirectory = file("$distDir/sources")
+    archiveFileName = "kotlin-endorsed-native-sources.zip"
+
+    includeEmptyDirs = false
+    include('**/*.kt')
+
+    from(project(':endorsedLibraries').file('build/sources-native'))
```

Comment:
I think referring anything in `build` directory is a dangerous practice. You may refer Gradle tasks, source sets, etc - any entity that Gradle knows how to build and knows how to check if it should be invalidated and re-built. But `build` directory can be treated as always holding stale data.

Reply:
> You may refer Gradle tasks, source sets, etc - any entity that Gradle knows how to build and knows how to check if it should be invalidated and re-built.

I think that using Gradle's dependency mechanism in the same manner as it's used for stdlib sources is the best choice. I've prototyped something similar for native sources: https://github.com/JetBrains/kotlin-native/commit/eb0f1b1489a479a93b534269e59659c35160129b.

Reply:
Thanks, but as Dmitry said that there should be separated zip archives for each library, there is no necessity in any coping, so I made all easier